library/core/src/iter/traits/iterator.rs

```diff
@@ -387,7 +392,9 @@ pub trait Iterator {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_do_not_const_check]
     fn nth(&mut self, n: usize) -> Option<Self::Item> {
-        self.advance_by(n).ok()?;
```

Comment:
don't need to change what you have, but this could be left as the previous thing, right?